packages/components/src/Combobox/Combobox.types.ts

```diff
@@ -34,10 +34,25 @@ export interface ComboboxProps {
    */
   readonly onClose?: () => void;
 
+  /**
+   * Debounced callback function invoked on Combobox search input change. Receives the current search value as an argument.
+   */
+  readonly onSearch?: (searchValue: string) => void;
+
+  /**
+   * The amount of time in ms to debounce the onSearch callback. Defaults to 300ms.
+   */
+  readonly onSearchDebounce?: number;
+
   /**
    * The Chip heading for the trigger
    */
   readonly label?: string;
+
+  /**
+   * Should the Combobox display the loading state, only needed when providing custom onSearch that has async loading behavior.
+   */
+  readonly loading?: boolean;
```

Comment:
> only needed when providing custom onSearch that has async loading behavior.

I think this shouldn't be tied to onSearch as initially loading is a valid case. Meaning, there is a query behind the first load of options and it is loading.

Real example in JO is it's loading the initial tags options in clients index.